test/common.sh

```diff
@@ -0,0 +1,276 @@
+function success() {
+  set_test_return_code 0
+  echo "**************************************"
+  echo "***        E2E TESTS PASSED        ***"
+  echo "**************************************"
+  exit 0
+}
+
+function show_failure() {
+    local testname=$1 tns=$2
+
+    echo "FAILED: ${testname} task has failed to comeback properly" ;
+    ${KUBECTL_CMD} api-resources
+    echo "Namespace: ${tns}"
+    echo "--- TaskRun Dump"
+    ${KUBECTL_CMD} get --ignore-not-found=true -n ${tns} taskrun -o yaml
+    echo "--- Task Dump"
+    ${KUBECTL_CMD} get --ignore-not-found=true -n ${tns} task -o yaml
+    echo "--- PipelineRun Dump"
+    ${KUBECTL_CMD} get --ignore-not-found=true -n ${tns} pipelinerun -o yaml
+    echo "--- Pipeline Dump"
+    ${KUBECTL_CMD} get --ignore-not-found=true -n ${tns} pipeline -o yaml
+    echo "--- StepAction Dump"
+    ${KUBECTL_CMD} get --ignore-not-found=true -n ${tns} stepaction -o yaml
+    echo "--- Container Logs"
+    for pod in $(${KUBECTL_CMD} get pod -o name -n ${tns}); do
+        echo "----POD_NAME: ${pod}---"
+        ${KUBECTL_CMD} logs --all-containers -n ${tns} ${pod} || true
+    done
+    exit 1
+
+}
+
+function test_yaml_can_install() {
+    # Validate that all the Task CRDs in this repo are valid by creating them in a NS.
+    ns="test-yaml-ns"
+    all_tasks="$*"
+    ${KUBECTL_CMD} create ns "${ns}" || true
+
+    local runtest
+    for runtest in ${all_tasks}; do
+        # remove task/ from beginning
+        local runtestdir=${runtest#*/}
+        # remove /0.1/tests from end
+        local testname=${runtestdir%%/*}
+        runtest=${runtest//tests}
+
+        # in case a task is being removed then it's directory
+        # doesn't exists, so skip the test for YAML
+        [ ! -d "${runtest%%/*}/${testname}" ] && continue
+
+        runtest="${runtest}${testname}.yaml"
+        skipit=
+        for ignore in ${TEST_YAML_IGNORES};do
+            [[ ${ignore} == "${testname}" ]] && skipit=True
+        done
+
+        # don't test the tasks which are deprecated
+        cat ${runtest} | grep 'tekton.dev/deprecated: \"true\"' && skipit=True
+
+        # In case if PLATFORM env variable is specified, do the tests only for matching tasks
+        [[ -n ${PLATFORM} ]] && [[ $(grep "tekton.dev/platforms" ${runtest} 2>/dev/null) != *"${PLATFORM}"* ]]  && skipit=True
+
+        [[ -n ${skipit} ]] && continue
+        echo "Checking ${testname}"
+        ${KUBECTL_CMD} -n ${ns} apply -f <(sed "s/namespace:.*/namespace: ${ns}/" "${runtest}")
+    done
+}
+
+function validate_task_yaml_using_tektor() {
+    all_tasks="$*"
+    for task in ${all_tasks}; do
+        # Skip if it is kustomize related yaml files
+        if [[ ${task} == *"kustomization.yaml" || ${task} == *"patch.yaml" || ${task} == *"recipe.yaml" ]]; then
+                continue
+        fi
+        tektor validate ${task}
+    done
+}
+
+function test_resource_creation() {
```

Comment:
The function name and what id does differs

Reply:
updated it to `test_resource_creation_and_validation`, hope the name fits with its functionality now